### DISCUSSION TITLE
Retrieving lower_ci, upper_ci for return periods

### DISCUSSION THREAD

**Discussion by mlicer** [georgebv/pyextremes#19] [Q&A] [answered]
Dear all,

i am sure there must be a simple way to retrieve `lower_ci` and `upper_ci` for return periods as two independent numpy arrays. I would like to do some custom plots and i don't seem to be able to obtain the confidence intervals...

I am currently doing something like:

```
...
return_periods = get_return_periods(
		ts=pd.Series(df[varname]),
		extremes=extremes,
		extremes_method=gevMethod,
		extremes_type="high",
		block_size=blockSize,
		return_period_size=blockSize,
		plotting_position="weibull",
	)

return_values = model.get_return_value(return_periods['return period'])

return eva_fit,extremes,return_periods, return_values
```

but all the returned `return_values` are `nan`...Any help would be appreciated!

Best regards,
M

**Comment by georgebv** [comment, marked as answer]
@mlicer This function returns a tuple with three arrays: `return_value`, `ci_lower`, and `ci_upper` as seen here:

https://github.com/georgebv/pyextremes/blob/f8b097fd49aede2253df23dc654488a540b728d4/src/pyextremes/eva.py#L1036-L1043

You can do the following:

```python
...
return_periods = get_return_periods(
		ts=pd.Series(df[varname]),
		extremes=extremes,
		extremes_method=gevMethod,
		extremes_type="high",
		block_size=blockSize,
		return_period_size=blockSize,
		plotting_position="weibull",
	)

reutrn_values, lower_ci, upper_ci = model.get_return_value(return_periods['return period'])
```

As for `nan`s, I would need to see your full code but my gut feeling is that your `blockSize` is not 1 year - `get_return_values` assumes that `return_period_size="1Y"` and if it is not that then you can get `nan`s when you ask for return period smaller than your block size in BM extremes.

**Reply by mlicer** [reply]
@georgebv Thank you very much for your reply and thanks for the library, i am sure it saved a lot of time to a lot of people! I have tried to modify the code along the lines of your suggestions but unfortunately i haven't been very successful. I've created a minimal example which reproduces the issue in this colab: 

[https://colab.research.google.com/drive/1q-HvazZvG6VB1Ii-so3_LIWvGi4Mr_2k?usp=sharing](https://colab.research.google.com/drive/1q-HvazZvG6VB1Ii-so3_LIWvGi4Mr_2k?usp=sharing)

If, by any chance, you can take a quick peek and execute the cells and let me know what i am doing wrong, i'd be most grateful! 
Best regards!

**Reply by georgebv** [reply]
@mlicer I have created a gist for you here https://gist.github.com/georgebv/cdb3f2ad2f8bd8e0ebac37cef430468d

The reason you were getting `nan`s was because you did not use the `alpha` parameter, which specifies confidence interval width. I should get it documented eventually.

**Reply by mlicer** [reply]
@georgebv thanks so much! This should clear things up. I truly appreciate it! 
All the best, M
